docs/lang/articles/advanced/odop2.md

```diff
@@ -0,0 +1,83 @@
+---
+sidebar_position: 3
+---
+
+# Objective Data-oriented Programming II
+
+
+## Taichi dataclasses
+
+Taichi provides custom [struct types](../type_system/type.md#compound-types) for developers to associate pieces of data together. However, it is often convenient to have:
```

Comment:
```suggestion
Taichi provides custom [struct types](../type_system/type.md#compound-types) for developers to assemble pieces of data together. However, it would be more convenient to have:
```